lucille-core/src/main/java/com/kmwllc/lucille/core/ExponentialBackoffRetryHandler.java

```diff
@@ -0,0 +1,39 @@
+package com.kmwllc.lucille.core;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import org.apache.http.client.HttpRequestRetryHandler;
+import org.apache.http.protocol.HttpContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ExponentialBackoffRetryHandler implements HttpRequestRetryHandler {
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  private final int maxNumRetries;
+  private final int initialExpiry;
+  private final int maxExpiry;
+
+  public ExponentialBackoffRetryHandler(int maxNumRetries, int initialExpiry, int maxExpiry) {
+    this.maxNumRetries = maxNumRetries;
+    this.initialExpiry = initialExpiry;
+    this.maxExpiry = maxExpiry;
+  }
+
+  @Override
+  public boolean retryRequest(IOException exception, int numRetries, HttpContext httpContext) {
+    if (numRetries > this.maxNumRetries) {
+      return false;
+    } else {
+      long delay = (long) Math.min(this.initialExpiry * Math.pow(2, numRetries - 1), this.maxExpiry);
+      log.info("Retrying request........waiting {}ms", delay);
```

Comment:
Should we find a way to include the URL in the log message? I'm thinking that if someone sees the current message in the logs, they'll have to do some piecing-together to figure out what request is actually being retried here, if that would even be possible.